Update experimental status on the component lifecycle (#394)
primer/design: content/guides/component-lifecycle.mdx

## Experimental

> The component exists as a proof-of-concept or is work in progress.

Author: I wasn't sure how much to write so only added the crucial information. Please feel free to suggest changes.

> - The component can be on any state on the maturity spectum from low to high. Please refer to the `status checklist in the component documentation` for fine-grained information about the component's maturity.

Author: We don't seem to have status checklist on `view-components` or maybe I missed that. I am not sure how inclusive this criteria 🤔 @primer/rails-reviewers

Author: Removed the bullet point so not relevant anymore.

> - The component can be on any state on the maturity spectum from low to high. Please refer to the `status checklist in the component documentation` for fine-grained information about the component's maturity.

Reviewer: ```suggestion
```

I might suggest removing this bullet point altogether. While it's possible to have a beta experimental component, I think this will be a rare use case and the idea of a mature experimental component is probably a bit challenging for non-maintainers to understand. Do you see any concerns with this suggested change?

Author: I see your point! I wanted to add this to encourage early adoptions or just trying out the component but now re-thinking it, the second bullet point does that job. So I think it is safe to remove this bullet point. Thank you!

> - The component can be ready for preliminary usage depending on its maturity. Please reach out to #primer to discuss further.

Reviewer: ```suggestion
- The component may be ready for preliminary usage depending on its maturity. Please reach out in [an issue](https://primer.style/contribute/how-to-contribute#contribute-to-our-open-source-repos) to discuss further.
```

Suggesting an issue over Slack since the component lifecycle is public to folks external to GitHub, and probably a good idea to keep a record of these conversations (via issues) anyway!

Author: Great point! ✨

## Experimental

The component exists as a proof-of-concept or is work in progress.

> - The component may be ready for preliminary usage depending on its maturity. Please reach out in [an issue](https://primer.style/contribute/how-to-contribute#contribute-to-our-open-source-repos) to discuss further.

Author: ```suggestion
- The component may be ready for preliminary usage depending on its maturity. Please file [an issue](https://primer.style/contribute/how-to-contribute#contribute-to-our-open-source-repos) to discuss further.
```
